### ISSUE TITLE
[C#] Implement new Concept Exercise: do-while-loops

### ISSUE THREAD

**Issue by mikedamay** [exercism/v3#1631]
This issue describes how to implement the `do-while-loops` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how and when to use  `do` keyword in C#.

## Learning objectives

- Know how to use `do` loops

## Out of scope

- Other loops

## Concepts

- `do-loops`

## Prerequisites

- `while-loops`

There may be other pre-requisites depending on the story.

### Hints

[do][do] documentation provices an introduction to `do` loops.

### After

[do][do] documentation provices an introduction to `do` loops.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[do]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/do 
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer

**Comment by mikedamay** [comment]
I am thinking of arrays of numbers received from some other system which need to be grouped and totaled.  Each group of numbers is terminated by a zero.  This is a stone cold fit for `do...while` but extremely artificial.   Using the more natural `char` runs the risk that the student will convert the array to a string and use `Split()`.  A `TextReader` or `Stream` would make things more natural but I did not want to make a big topic like that a dependency for introductory material like this.

Any comments would be welcome.

**Comment by ErikSchierboom** [comment]
> I am thinking of arrays of numbers received from some other system which need to be grouped and totaled. Each group of numbers is terminated by a zero. This is a stone cold fit for `do...while` but extremely artificial. Using the more natural `char` runs the risk that the student will convert the array to a string and use `Split()`. A `TextReader` or `Stream` would make things more natural but I did not want to make a big topic like that a dependency for introductory material like this.
> 
> Any comments would be welcome.

I agree with everything said here. If we would use `char` we could use an analyzer to verify that the student used a `do...while` loop. That said, an array of numbers could work, if we maybe come up with a nice-ish story around it. Maybe we could have the array of numbers be some sort of communication or encryption protocol?

**Comment by mikedamay** [comment]
Closed by #1737 which adds the `do-while` concept to `floating-point-numbers` where `while-loops` are a concept as a discussion point only.
